modules/base/rendering/grids/renderableradialgrid.cpp

```diff
@@ -0,0 +1,461 @@
+        glDisablei(GL_BLEND, 0);
+    }
+
+    if (!isLineSmoothEnabled) {
+        glDisable(GL_LINE_SMOOTH);
+    }
+}
+
+void RenderableRadialGrid::update(const UpdateData&) {
+    if (_gridIsDirty) {
+
+        // Circles
+        const int nRadialSegments = _gridSegments.value()[0];
+        const float fnCircles = static_cast<float>(nRadialSegments);
+        const float deltaRadius = (_maxRadius - _minRadius) / fnCircles;
+
+        const bool hasInnerRadius = _minRadius > 0;
+        const int nCircles = hasInnerRadius ? nRadialSegments : nRadialSegments + 1;
+
+        _circles.clear();
+        _circles.reserve(nCircles);
+
+        auto addRing = [&](int nSegments, float radius) {
+            std::vector<rendering::helper::Vertex> vertices =
+                rendering::helper::createRing(nSegments, radius);
+
+            _circles.push_back(GeometryData(GL_LINE_STRIP));
+            _circles.back().varray = rendering::helper::convert(vertices);
+            _circles.back().update();
+        };
+
+        // add an extra inmost circle
+        if (hasInnerRadius) {
+            addRing(_circleSegments, _minRadius);
+        }
+
+        for (int i = 0; i < nRadialSegments; ++i) {
+            float ri = static_cast<float>(i + 1) * deltaRadius;
+            ri += _minRadius;
+            addRing(_circleSegments, ri);
+        }
+
+        // Lines
+        const int nLines = _gridSegments.value()[1];
+        const int nVertices = 2 * nLines;
+        const float fsegments = static_cast<float>(nLines);
+
+        _lines.varray.clear();
+        _lines.varray.reserve(nVertices);
+
+        if (nLines > 1) {
+            for (int i = 0; i < nLines; ++i) {
+                const float fi = static_cast<float>(i);
+
+                const float theta = fi * glm::pi<float>() * 2.0f / fsegments;  // 0 -> 2*PI
+
+                float x = _maxRadius * cos(theta);
+                float y = _maxRadius * sin(theta);
+                float z = 0.0f;
+
+                _lines.varray.push_back({ x, y, z });
+
+                x = _minRadius * cos(theta);
+                y = _minRadius * sin(theta);
+
+                _lines.varray.push_back({ x, y, z });
+            }
+        }
+        _lines.update();
+
+        _gridIsDirty = false;
+    }
+}
+
+RenderableRadialGrid::GeometryData::GeometryData(GLenum renderMode) 
+    : mode(renderMode)
+{
+    glGenVertexArrays(1, &vao);
+    glGenBuffers(1, &vbo);
+
+    glBindVertexArray(vao);
+    glBindBuffer(GL_ARRAY_BUFFER, vbo);
+    glEnableVertexAttribArray(0);
+    glBindVertexArray(0);
+}
+
+RenderableRadialGrid::GeometryData::GeometryData(GeometryData&& other) noexcept {
+    if (this == &other) return;
+    vao = other.vao;
+    vbo = other.vbo;
+    varray = std::move(other.varray);
+    mode = other.mode;
+
+    other.vao = 0;
+    other.vbo = 0;
+}
+
+RenderableRadialGrid::GeometryData& 
+RenderableRadialGrid::GeometryData::operator=(GeometryData&& other) noexcept {
```

Comment:
In this case, the `{` would go onto a new line, if the function definition doesn't fit on one line